### PR TITLE
Expose metrics

### PR DESCRIPTION
**Pull request by dwelch0**
https://issues.redhat.com/browse/APPSRE-6689

Implement standardized metrics for monitoring app-interface related automation

mirrors implementation here: https://github.com/app-sre/vault-manager/pull/81

### CODE REVIEW

File: run.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    -e AWS_SECRET_ACCESS_KEY=$AWS_SECRET_ACCESS_KEY \
+    -e AWS_REGION=$AWS_REGION \
+    -e AWS_S3_BUCKET=$AWS_S3_BUCKET \
+    -e GITLAB_BASE_URL=$GITLAB_BASE_URL\
+    -e GITLAB_USERNAME=$GITLAB_USERNAME \
+    -e GITLAB_TOKEN=$GITLAB_TOKEN \
+    -e PUBLIC_KEY=$PUBLIC_KEY \
+    -e RECONCILE_SLEEP_TIME=$RECONCILE_SLEEP_TIME \
+    -e WORKDIR=$WORKDIR \
+    quay.io/app-sre/git-partition-sync-producer:latest -dry-run=true -run-once=false
```
Comment (TGPSKI) [inline run.sh:18]: newline

File: pkg/utils/metrics.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	reconcileSuccessCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "qontract_reconcile_execution_counter_total",
```
Comment (TGPSKI) [inline pkg/utils/metrics.go:12]: good pattern matching vm

File: main.go [hunk 3]
```diff
@@ -40,16 +45,27 @@ func main() {
...
 	}
 
+	// retrieve raw from graphql
+
 	for {
-		ctx := context.Background()
+		status := 0
+		start := time.Now()
 
+		ctx := context.Background()
```
Comment (TGPSKI) [inline main.go:68]: Tell me more about this

Comment (dwelch0) [reply main.go:68]: on each iteration (reconcile loop), a new context is created. This is later used to [derive a timeout context](https://github.com/app-sre/git-partition-sync-producer/blob/master/pkg/uploader.go#L212) when querying the graphql server

Comment (TGPSKI) [reply main.go:68]: Actually I see its not a change.

pass the context into the http handlers and have a way to exit gofuncs nested in handlers

status/start needed for metrics

right?

Comment (dwelch0) [reply main.go:68]: correct. status/start for metrics. main ctx part is currently for setting deadline on graphql query. Certainly room for further refactors that utilize ctx

**Pull request merged**